File: asignaturas/sql/Asignaturas-IvanRonco/Profesores_Y_Modulos_BD-Java/src/main/java/mapper/profesor/ProfesorMapper.java
package mapper.profesor;

import mapper.modulo.ModuloMapper;
import dto.profesor.ListaProfesoresDto;
import dto.profesor.ProfesorDto;
import models.Profesor;

import java.time.LocalDateTime;
import java.util.List;

import static mapper.modulo.ModuloMapper.toModulo;

public class ProfesorMapper {
    public static ProfesorDto toProfesorDto(Profesor profesor){
        return new ProfesorDto(
                String.valueOf(profesor.getId()),
                profesor.getNombre(),
                profesor.getFechaIncorporacion().toString(),
                ModuloMapper.toModulosDto(profesor.getModulos()).getModuloDtos()
        );
    }

    public static Profesor toProfesor(ProfesorDto profesorDto){
        return new Profesor(
                Long.parseLong(profesorDto.getId()),
                profesorDto.getNombre(),
                LocalDateTime.parse(profesorDto.getFechaIncorporacion()),
                profesorDto.getModulosDto().stream().map(it -> toModulo(it)).toList()
        );
    }

    public static ListaProfesoresDto toProfesoresDto(List<Profesor> profesors){
        return new ListaProfesoresDto(
                profesors.stream().map(it -> toProfesorDto(it)).toList()
        );
    }

    public static List<Profesor> toProfesores(ListaProfesoresDto profesorsDto){
        return profesorsDto.getProfesoresDtos().stream().map(it -> toProfesor(it)).toList();
    }
}

File: asignaturas/sql/Modulo-Profesor_OscarGrc/src/main/kotlin/mappers/ProfesorMapper.kt
package mappers
import Dto.ProfesorDTO
import models.*
import java.time.LocalDate

fun Profesor.toDTO(): ProfesorDTO {
        return ProfesorDTO(id = this.id.toString(),
            nombre = this.nombre, fechaIncorporacion = this.fehcaIncorpracion.toString())
}

fun ProfesorDTO.toCoche(): Profesor {
        return Profesor(
            id = this.id.toLong(),
            nombre = this.nombre, fehcaIncorpracion = LocalDate.parse(this.fechaIncorporacion))

}